modules/core/src/test/java/org/apache/ignite/internal/processors/security/cache/CacheOperationPermissionCheckTest.java

```diff
@@ -23,51 +23,106 @@
 import static org.apache.ignite.plugin.security.SecurityPermission.CACHE_REMOVE;
+import static org.apache.ignite.plugin.security.SecurityPermission.JOIN_AS_SERVER;
 import static org.apache.ignite.testframework.GridTestUtils.assertThrowsWithCause;
 
 /**
  * Test CRUD cache permissions.
  */
 @RunWith(JUnit4.class)
-public class CacheOperationPermissionCheckTest extends AbstractCacheOperationPermissionCheckTest {
-    /** */
+public class CacheOperationPermissionCheckTest extends AbstractSecurityTest {
+    /** New cache for tests. */
+    protected static final String NEW_CACHE = "NEW_CACHE";
+
+    /** Cache name for tests. */
+    protected static final String ALL_PERM_TEST_CACHE = "ALL_PERM_TEST_CACHE";
+
+    /** Forbidden caches. */
+    protected static final String CREATE_TEST_CACHE = "CREATE_TEST_CACHE";
+    protected static final String EMPTY_PERM_TEST_CACHE = "EMPTY_PERM_TEST_CACHE";
+
+    /**
+     *
+     */
     @Test
-    public void testServerNode() throws Exception {
+    public void testServerCrudCacheNode() throws Exception {
         testCrudCachePermissions(false);
     }
 
-    /** */
+    /**
+     *
+     */
     @Test
-    public void testClientNode() throws Exception {
+    public void testClientCrudCacheNode() throws Exception {
         testCrudCachePermissions(true);
     }
 
+    /**
+     *
+     */
+    @Test
+    public void testServerCrudCacheSystemNode() throws Exception {
+        testCrudCacheSystemPermissions(false);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testClientCrudCacheSystemNode() throws Exception {
+        testCrudCacheSystemPermissions(true);
+    }
+
     /**
      * @param isClient True if is client mode.
      * @throws Exception If failed.
      */
     private void testCrudCachePermissions(boolean isClient) throws Exception {
-        Ignite node = startGrid(loginPrefix(isClient) + "_test_node",
-            SecurityPermissionSetBuilder.create()
-                .appendCachePermissions(CACHE_NAME, CACHE_READ, CACHE_PUT, CACHE_REMOVE)
-                .appendCachePermissions(FORBIDDEN_CACHE, EMPTY_PERMS).build(), isClient);
+        String login = isClient ? "client" : "server";
+        Ignite node = startGrid(login + "_test_node",
+            getSecurityPermissionSet(JOIN_AS_SERVER),
+            isClient);
 
-        for (Consumer<IgniteCache<String, String>> c : operations()) {
-            c.accept(node.cache(CACHE_NAME));
+        node.createCache(NEW_CACHE); // if defaultAllowAll == false, there will be exeption
+        node.createCache(ALL_PERM_TEST_CACHE);
+        node.createCache(CREATE_TEST_CACHE);
 
-            assertThrowsWithCause(() -> c.accept(node.cache(FORBIDDEN_CACHE)), SecurityException.class);
-        }
+        assertThrowsWithCause(() -> node.createCache(EMPTY_PERM_TEST_CACHE), SecurityException.class);
+
+        checkOperations(node);
+
+        assertThrowsWithCause(() -> node.cache(CREATE_TEST_CACHE).destroy(), SecurityException.class);
```

Comment:
What about other caches? We should also check here wheter their destruction succeeds/fails.